pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -1558,8 +1565,62 @@ func (volTopology *wcpControllerVolumeTopology) GetTopologyInfoFromNodes(ctx con
 
 	switch strings.ToLower(params.StorageTopologyType) {
 	case "zonal":
-		// If the topology requirement received has just one zone, use the same zone as node affinity terms on PV.
-		if len(params.TopologyRequirement.GetPreferred()) == 1 {
+		if params.TopologyRequirement == nil {
+			// If the topology requirement received is nil, then identigy topology of the datastore by looking into
+			// azClustersMap
+			if !isPodVMOnStretchedSupervisorEnabled {
+				return nil, logger.LogNewErrorf(log, "invalid params: %v", params)
+			} else {
+				var selectedSegments []map[string]string
+				for az, clusters := range azClustersMap {
+					var sharedDatastoresForclusterMorefs []*cnsvsphere.DatastoreInfo
+					for index, clusterMoref := range clusters {
+						accessibleDs, _, err := cnsvsphere.GetCandidateDatastoresInCluster(ctx, params.Vc, clusterMoref)
+						if err != nil {
+							return nil, logger.LogNewErrorf(log,
+								"failed to find candidate datastores to place volume in cluster %q. Error: %v",
+								clusterMoref, err)
+						}
+						if index == 0 {
+							sharedDatastoresForclusterMorefs = append(sharedDatastoresForclusterMorefs, accessibleDs...)
+						} else {
+							var sharedAccessibleDatastores []*cnsvsphere.DatastoreInfo
+							for _, sharedDatastore := range sharedDatastoresForclusterMorefs {
+								for _, accessibleDsInCluster := range accessibleDs {
+									if sharedDatastore.Info.Url == accessibleDsInCluster.Info.Url {
+										sharedAccessibleDatastores = append(sharedAccessibleDatastores, accessibleDsInCluster)
+										break
+									}
+								}
+							}
+							sharedDatastoresForclusterMorefs = sharedAccessibleDatastores
+						}
+					}
+					for _, ds := range sharedDatastoresForclusterMorefs {
+						if ds.Info.Url == params.DatastoreURL {
+							selectedSegments = append(selectedSegments, map[string]string{"topology.kubernetes.io/zone": az})
```

Comment:
Use `LabelTopologyZone` constant from core v1 upstream package instead of `topology.kubernetes.io/zone`

Reply:
done